src/wallet/wallet.cpp

```diff
@@ -4167,11 +4167,11 @@ CWallet* CWallet::CreateWalletFromFile(const std::string& name, const fs::path&
         nStart = GetTimeMillis();
         {
             WalletRescanReserver reserver(walletInstance);
-            if (!reserver.reserve()) {
+            const CBlockIndex* stop_block = nullptr;
+            if (!reserver.reserve() || !walletInstance->ScanForWalletTransactions(pindexRescan, nullptr, stop_block, reserver, true)) {
```

Comment:
Note this did not previously fail when ScanForWalletTransactions failed, so this removes a silent failure.

Reply:
I think this is ok. This just adds another case where `walletInstance` is leaked on error.

Reply:
Should we be switching `walletInstance` to RAII then?